aquacomputer_d5next.c

```diff
@@ -2021,7 +2051,13 @@ static int aqc_probe(struct hid_device *hdev, const struct hid_device_id *id)
 
 		priv->num_temp_sensors = AQUASTREAMXT_NUM_SENSORS;
 		priv->temp_sensor_start_offset = AQUASTREAMXT_SENSOR_START;
-		priv->buffer_size = AQUASTREAMXT_SENSOR_REPORT_SIZE;
+
+		/*
+		 * Since we use the same buffer for both sensor and control
+		 * report storage on legacy devices, reserve enough space
+		 */
+		priv->buffer_size = max(AQUASTREAMXT_SENSOR_REPORT_SIZE,
+					AQUASTREAMXT_CTRL_REPORT_SIZE);
```

Comment:
Nice, i didn't think of this before. Maybe we should use separate buffers for sensor and control reports in the future.

Reply:
I thought about that as well. I think it's not necessary since devices that communicate this way would then need both buffers if reading and writing is implemented (and because we clear it every time something is written or read, it just needs to be big enough).